internal/devbox/providers/nixcache/nixcache.go

```diff
@@ -0,0 +1,99 @@
+package nixcache
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cognitoidentity/types"
+	"go.jetpack.io/devbox/internal/build"
+	"go.jetpack.io/devbox/internal/devbox/providers/identity"
+	"go.jetpack.io/devbox/internal/nix"
+	"go.jetpack.io/devbox/internal/ux"
+	"go.jetpack.io/pkg/api"
+	"go.jetpack.io/pkg/auth"
+)
+
+type Provider struct{}
+
+var singleton *Provider = &Provider{}
```

Comment:
why do providers need to be singletons?

Reply:
They don't right now, but I'm hoping to add caching to avoid making multiple API calls. Will do this after testing end to end and understanding where the perf bottlenecks are.